lib/galaxy/webapps/galaxy/api/history_contents.py

```diff
@@ -1105,8 +1119,10 @@ def contents_near(self, trans, history_id, hid, limit, **kwd):
         trans.response.headers['total_matches_down'] = down_count
         trans.response.headers['max_hid'] = max_hid
         trans.response.headers['min_hid'] = min_hid
+        trans.response.headers['history_size'] = str(history.disk_size)
+        trans.response.headers['history_empty'] = json.dumps(history.empty)  # convert to proper bool
```

Comment:
history.empty is `history.hid_counter == 1` internally, where hid counter is a column on history that indicates the next hid to be used when adding datasets or collections. If you have any use for the total hid count it might be better to expose that ?

Reply:
I'm trying to get the history summary query to tell me whether the histories are empty or not. I don't mind making "empty" a computation based on hid_counter, but you're the guy with 10,000 histories.

How's this query run on your machine?
http://localhost:8081/api/histories?view=summary&keys=hid_counter

Or this?
http://localhost:8081/api/histories?view=summary&keys=empty

The whole point of this is that you said the history list runs slow, so I'm back to the basic summary query.

Reply:
Ahh you know what, nevermind. That'll be a tiny change, we can get to it later.